doc/HD44780.rst

```diff
@@ -0,0 +1,197 @@
+the display is to use the `text` property which operates similarly to the
+:py:class:`luma.core.virtual.sevensegment` wrapper class.  To use it, assign `text`
+a string containing the values that you want displayed.
+
+.. code:: python
+
+  from luma.core.interface.serial import parallel
+  from luma.lcd.device import hd44780
+
+  p = parallel(RS=7, E=8, PINS=[25,24,23,27])
+  d = hd44780(p)
+
+  d.text = 'Hello World'
+
+The newline character can be used to display characters on the next line of the
+display.
+
+.. code:: python
+
+  d.text = 'Hello\nWorld'
+
+It is the only control character that works however.  Other control characters
+such as carriage return (ascii 13) and line feed (ascii 10) will be silently
+ignores.
+
+.. note:
+  Unlike sevensegment's text property which throws an exception if you exceed
+  the bounds of the display, there is no bounds checking on the displayed text.
+  If you exceed a line boundary it will not wrap to the next line and content
+  below the lowest line on the device will not be displayed.
+
+Graphics-mode Usage
+^^^^^^^^^^^^^^^^^^^
+It is also possible to display graphical content using the `display` method
+which works similarly to other luma.lcd and luma.oled displays.  However there is
+a significant limitation.  There can only be 8 characters worth of 'custom'
+content displayed at any point in time.  While this is quite limited, with some
+creativity it can be very useful.
+
+To understand how to leverage this capability requires some explanation of
+how the screen of the display is organized and how luma.lcd manages the
+custom character space.
+
+Each character position on an HD44780 display is made up of a 5 pixel by 8 pixel
+grid.  When a character is requested to be displayed, the HD44780 looks up the
+character from its font table and copies the pixels from the font table ROM to the
+address in RAM that corresponds to the current character position.  If there is
+not a character within the font table that matches what needs to be displayed,
+a custom character can be created within a small space in the display's RAM that
+is reserved for this purpose.  There are several restrictions that have to be
+kept in mind though.
+
+* The custom character must be a 5x8 image
+* It can only be displayed in alignment with the other characters on the display
+* There can be a maximum of 8 special characters on the screen at any given time
+
+However there are two features that can be leveraged to extend beyond these
+restrictions.
+
+* The content of a special character can be used multiple times on the screen
+* The content of all of the special characters can be changed every time the
+  screen is redrawn.
+
+Here is a small example of how this can be leveraged.
+
+.. code:: python
+
+
+  from luma.core.interface.serial import parallel
+  from luma.lcd.device import hd44780
+  from luma.core.render import canvas
+  from PIL import Image, ImageDraw
+
+  serial = parallel(RS=7, E=8, PINS=[25,24,23,27])
```

Comment:
Potentially confusing name for a parallel device

Reply:
Changed the examples in HD44780.rst to use less confusing variable names.